src/gui/processing/qgsprocessingalgorithmdialogbase.cpp

```diff
@@ -592,6 +592,14 @@ void QgsProcessingAlgorithmDialogBase::setInfo( const QString &message, bool isE
   processEvents();
 }
 
+void QgsProcessingAlgorithmDialogBase::reject()
+{
+  if ( !mAlgorithmTask ){
+    setAttribute( Qt::WA_DeleteOnClose );
+  }
+  done( QDialog::Rejected );
```

Comment:
I think this should be `QDialog::reject()`

Reply:
Fixed, rebased and pushed in https://github.com/qgis/QGIS/pull/34796/commits/0bb5c6f0a71520fdb6844789b28ee9ea9a050ec4